### PR TITLE
deprecate util.frame

### PR DESCRIPTION
**Pull request by mustafa0x**
Funny that this is even in. The README says:

> wavesurfer.js works only in modern browsers supporting Web Audio.

Anything that supports Web Audio supports rAF.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
thanks @mustafa0x, can you also add a changelog entry?

**Comment by thijstriemstra** [comment]
Forgot to mention; the other deprecation you added can also be added to the changelog.

File: CHANGES.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ wavesurfer.js changelog
   - Disable drag selection before enabling it (#1698)
   - Add `channelIdx` option to select specific channel to draw on (#1829)
 - Cursor plugin: fix time visibility (#1802)
+- Deprecate `util.frame` (i.e. rAF shim) (#1833)
```
Comment (thijstriemstra) [inline CHANGES.md]: i would use full name instead of rAF

File: src/util/request-animation-frame.js [hunk 2]
```diff
@@ -11,5 +11,5 @@ export default (
     window.mozRequestAnimationFrame ||
     window.oRequestAnimationFrame ||
     window.msRequestAnimationFrame ||
-    ((callback, element) => setTimeout(callback, 1000 / 60))
```
Comment (thijstriemstra) [inline src/util/request-animation-frame.js]: no need to change the existing implementation.

File: src/util/frame.js [hunk 3]
```diff
@@ -9,5 +7,8 @@ import reqAnimationFrame from './request-animation-frame';
  * @return {func} The function wrapped within a requestAnimationFrame
  */
 export default function frame(func) {
-    return (...args) => reqAnimationFrame(() => func(...args));
+    console.warn(
+        'util.frame is deprecated; use window.requestAnimationFrame instead'
+    );
+    return (...args) => window.requestAnimationFrame(() => func(...args));
```
Comment (thijstriemstra) [inline src/util/frame.js]: no need to change the existing implementation.

File: src/util/request-animation-frame.js (hunk 2 above)
Comment (mustafa0x) [reply src/util/request-animation-frame.js]: Don't be anal man 😂 

It was an unused arg.

Comment (thijstriemstra) [reply src/util/request-animation-frame.js]: Love it or hate it, I don't care.

Comment (mustafa0x) [reply src/util/request-animation-frame.js]: Mostly bemusement at the insistence for keeping an unused arg around.

Comment (mustafa0x) [reply src/util/request-animation-frame.js]: This isn't changing implementation, it's removing an unused arg. 🤷‍♂ 

Nothing else changed.

File: src/util/frame.js (hunk 3 above)
Comment (mustafa0x) [reply src/util/frame.js]: 😂 we should keep the dep on a deprecated module? 😂

File: src/util/request-animation-frame.js (hunk 2 above)
Comment (thijstriemstra) [reply src/util/request-animation-frame.js]: There's a lot of quirks in old code, who knows, maybe it's used in some weird way somewhere, by someone. If we keep it, that will definitely **not** cause issues. No need to introduce a change to this function.

File: src/util/frame.js (hunk 3 above)
Comment (thijstriemstra) [reply src/util/frame.js]: What's up with the 😂 😂 ? I am trying to give you guidance and you think it's funny? Why the disrespect?

Comment (mustafa0x) [reply src/util/frame.js]: Disrespect?! Not at all! =)

**Comment by mustafa0x** [comment]
Well, this was fun! =)

I'll excuse myself =)

File: src/util/frame.js (hunk 3 above)
Comment (thijstriemstra) [reply src/util/frame.js]: And what does that even mean 'the dep on a deprecated module'? You can replace usage of util.frame in the code and simply leave util.frame alone, besides adding a deprecation warning.

File: src/util/request-animation-frame.js (hunk 2 above)
Comment (mustafa0x) [reply src/util/request-animation-frame.js]: In a language like JS that's literally impossible. =)

**Pull request closed without merging**